server/src/main/java/io/druid/server/coordinator/rules/LoadRule.java

```diff
@@ -57,12 +59,58 @@ public CoordinatorStats run(DruidCoordinator coordinator, DruidCoordinatorRuntim
 
     final Map<String, Integer> loadStatus = Maps.newHashMap();
 
+    final Map<String, Integer> tieredReplicants = getTieredReplicants();
+    for (final String tier : tieredReplicants.keySet()) {
+      stats.addToTieredStat(ASSIGNED_COUNT, tier, 0);
```

Comment:
why does this need initialized now?

Reply:
I don't think this needs initialized. All the components underneath lazily add entries as needed

Reply:
This is for the expected behavior in once of the tests: https://github.com/druid-io/druid/blob/d43687d578bf3dea98b01d0899bcfbb2125d142e/server/src/test/java/io/druid/server/coordinator/rules/LoadRuleTest.java#L631

Without the initialization, `stat3` would not have the `hot` tier and throw NPE. Note that previously, a `0` will be added into the statistics when no assignment is done:
https://github.com/druid-io/druid/blob/d43687d578bf3dea98b01d0899bcfbb2125d142e/server/src/main/java/io/druid/server/coordinator/rules/LoadRule.java#L102

Personally, I also feel that no initialization is required but I am not sure if receiving `0` is part of the expected usage.